handler/src/main/java/io/netty/handler/pcap/PcapWriteHandler.java

```diff
@@ -0,0 +1,523 @@
+     */
+    private final boolean writePcapGlobalHeader;
+
+    /**
+     * TCP Sender Segment Number.
+     * It'll start with 1 and keep incrementing with number of bytes read/sent.
+     */
+    private int sendSegmentNumber = 1;
+
+    /**
+     * TCP Receiver Segment Number.
+     * It'll start with 1 and keep incrementing with number of bytes read/sent.
+     */
+    private int receiveSegmentNumber = 1;
+
+    /**
+     * Source Address
+     */
+    private InetSocketAddress srcAddr;
+
+    /**
+     * Destination Address
+     */
+    private InetSocketAddress dstAddr;
+
+    /**
+     * Create new {@link PcapWriteHandler} Instance.
+     * {@code captureZeroByte} is set to {@code false} and
+     * {@code writePcapGlobalHeader} is set to {@code true}.
+     *
+     * @param outputStream OutputStream where Pcap data will be written
+     * @throws NullPointerException If {@link OutputStream} is {@code null} then we'll throw an
+     *                              {@link NullPointerException}
+     */
+    public PcapWriteHandler(OutputStream outputStream) {
+        this(outputStream, false, true);
+    }
+
+    /**
+     * Create new {@link PcapWriteHandler} Instance
+     *
+     * @param outputStream          OutputStream where Pcap data will be written
+     * @param captureZeroByte       Set to {@code true} to enable capturing packets with empty (0 bytes) payload.
+     *                              Otherwise, if set to {@code false}, empty packets will be filtered out.
+     * @param writePcapGlobalHeader Set to {@code true} to write Pcap Global Header on initialization.
+     *                              Otherwise, if set to {@code false}, Pcap Global Header will not be written
+     *                              on initialization. This could when writing Pcap data on a existing file where
+     *                              Pcap Global Header is already present.
+     * @throws NullPointerException If {@link OutputStream} is {@code null} then we'll throw an
+     *                              {@link NullPointerException}
+     */
+    public PcapWriteHandler(OutputStream outputStream, boolean captureZeroByte, boolean writePcapGlobalHeader) {
+        this.outputStream = ObjectUtil.checkNotNull(outputStream, "OutputStream");
+        this.captureZeroByte = captureZeroByte;
+        this.writePcapGlobalHeader = writePcapGlobalHeader;
+    }
+
+    @Override
+    public void channelActive(ChannelHandlerContext ctx) throws Exception {
+
+        ByteBufAllocator byteBufAllocator = ctx.alloc();
+
+        /*
+         * If `writePcapGlobalHeader` is `true`, we'll write Pcap Global Header.
+         */
+        if (writePcapGlobalHeader) {
+
+            ByteBuf byteBuf = byteBufAllocator.buffer();
+            try {
+                this.pCapWriter = new PcapWriter(this.outputStream, byteBuf);
+            } catch (IOException ex) {
+                ctx.channel().close();
+                ctx.fireExceptionCaught(ex);
```

Comment:
Is it make sense to send ctx.fireExceptionCaught(ex) after we close channel ? I think it reach only `TailContext`

Reply:
We need to call `ctx.fireExceptionCaught(ex)` because an `IOException` is caught due to `OutputStream`. So we'll close the channel since `OutputStream` has errors and writing packets won't work.